src/cargo/util/toml/mod.rs

```diff
@@ -1320,43 +1320,43 @@ impl TomlManifest {
         config: &Config,
     ) -> CargoResult<(VirtualManifest, Vec<PathBuf>)> {
         if me.project.is_some() {
-            bail!("virtual manifests do not define [project]");
+            bail!("This virtual manifest specifies a [project] section, which is not allowed");
```

Comment:
Currently all of Cargo's error messages conventionally start with a lowercase letter, mind changing that here? Other than that this seems good to go!

Reply:
Done.